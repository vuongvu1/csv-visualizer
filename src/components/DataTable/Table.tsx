import React from "react";
import SC from "./styles";

interface Props {
  headerRow: string[];
  showingRows: Array<string[]>;
}

// have to use indexes as keys because rows don't have ids or any unique values
const Table: React.FC<Props> = ({ headerRow, showingRows }) => (
  <SC.Table>
    <thead>
      <SC.Tr>
        {headerRow.map((header, index) => (
          <SC.Th key={index}>{header}</SC.Th>
        ))}
      </SC.Tr>
    </thead>
    <tbody>
      {showingRows.map((row, index) => (
        <SC.Tr key={index}>
          {row.map((cell, ind) => (
            <SC.Td key={ind}>{cell}</SC.Td>
          ))}
        </SC.Tr>
      ))}
    </tbody>
  </SC.Table>
);

export default Table;
