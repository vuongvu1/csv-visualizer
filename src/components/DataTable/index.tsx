import React, { useState } from "react";
import SC from "./styles";
import Pagination from "../Pagination";

interface Props {}

const DataTable: React.FC<Props> = () => {
  const [currentPage, setCurrentPage] = useState(1);

  return (
    <SC.Container>
      <Pagination
        total={10}
        current={currentPage}
        onChange={(_, nextPage) => setCurrentPage(nextPage)}
      />
    </SC.Container>
  );
};

export default DataTable;
