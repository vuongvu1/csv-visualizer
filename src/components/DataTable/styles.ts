import styled from "styled-components";
import { Spacing } from "configs/styles";
import {
  whiteColor,
  boxShadowColor,
  neutralColor,
  secondaryLightColor,
  loadingOverlayColor,
} from "configs/colors";

export default {
  Container: styled.div`
    flex: 2;
    display: flex;
    flex-direction: column;
    margin: ${Spacing.large};
    padding: ${Spacing.normal};
    background-color: ${whiteColor};
    box-shadow: 0px 1px 0px ${boxShadowColor};
    border-radius: 12px;
  `,
  TableContainer: styled.div`
    width: 100%;
    max-width: 552px;
    position: relative;
    margin: ${Spacing.normal} 0;
  `,
  Table: styled.table`
    display: block;
    border-collapse: collapse;
    width: 100%;
    overflow-x: auto;
    white-space: nowrap;
  `,
  Th: styled.th`
    border: 1px solid ${neutralColor};
    text-align: left;
    padding: ${Spacing.small};
  `,
  Td: styled.td`
    border: 1px solid ${neutralColor};
    text-align: left;
    padding: ${Spacing.small};
    height: 36px;
  `,
  Tr: styled.tr`
    &:nth-child(even) {
      background-color: ${secondaryLightColor};
    }
  `,
  LoadingOverlay: styled.div`
    position: absolute;
    top: 0;
    bottom: 0;
    left: 0;
    right: 0;
    background-color: ${loadingOverlayColor};
    display: flex;
    justify-content: center;
    align-items: center;
  `,
};
