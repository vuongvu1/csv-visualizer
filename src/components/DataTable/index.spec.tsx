import React from "react";
import { render, cleanup } from "@testing-library/react";
import DataTable from "./index";

describe("DataTable", () => {
  afterEach(cleanup);

  test("should match snapshot", () => {
    const { container } = render(<DataTable data={[]} />);
    expect(container).toMatchSnapshot();
  });
});
