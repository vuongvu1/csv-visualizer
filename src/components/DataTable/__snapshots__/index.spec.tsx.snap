// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DataTable should match snapshot 1`] = `
.c0 {
  -webkit-flex: 2;
  -ms-flex: 2;
  flex: 2;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  margin: 18px;
  padding: 12px;
  background-color: #FFFFFF;
  box-shadow: 0px 1px 0px rgba(165,165,168,0.2);
  border-radius: 12px;
}

<div>
  <div
    class="c0"
  >
    No data to display.
  </div>
</div>
`;
